app/step-functions-templates/run_qc_stats_sfn_template.asl.json

```diff
@@ -58,13 +58,46 @@
       "Type": "Choice",
       "Choices": [
         {
-          "Next": "Decompress Fastqs",
+          "Next": "ReadCount is null",
           "Condition": "{% $fastqObj.readSet.compressionFormat = 'ORA' %}",
           "Comment": "Is ORA Compression Format"
         }
       ],
       "Default": "Calculate Ephemeral Storage Size"
     },
+    "ReadCount is null": {
+      "Type": "Choice",
+      "Choices": [
+        {
+          "Next": "Calculate ReadCount",
+          "Condition": "{% $fastqObj.readCount = null %}"
+        }
+      ],
+      "Default": "Decompress Fastqs"
+    },
+    "Calculate ReadCount": {
+      "Type": "Task",
+      "Resource": "arn:aws:states:::events:putEvents.waitForTaskToken",
+      "Arguments": {
+        "Entries": [
+          {
+            "Detail": {
+              "taskToken": "{% $states.context.Task.Token %}",
+              "payload": {
+                "fastqIdList": ["{% $fastqId %}"],
+                "requirements": {
+                  "hasReadCountInformation": true
+                }
+              }
+            },
+            "DetailType": "${__fastq_sync_detail_type__}",
+            "EventBusName": "${__event_bus_name__}",
+            "Source": "${__stack_source__}"
+          }
+        ]
+      },
+      "Next": "Get fastq object"
```

Comment:
Shouldn't the next step be `"Decompress Fastqs"`?
And looks like `"Get fastq object"` does not exist?

Reply:
Ah, it's a loop back! Got it, thanks!